### PR TITLE
merging 2 rules into 1

### PR DESCRIPTION
**Pull request by parmeet4dev**
Merged two ****byte match set **** rules to one using the HCL2 language by replacing the strings with the Map of lists.
An example is updated on the homepage usage section

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,11 +20,27 @@ Example ACL to add a rate limit to the route `/session` of a given ALB (The ALB
...
+    }
+  ]
+}
 module "rate_based_rule" {
   source                = "github.com/ryte/INF-tf-waf.git//rate_based_rule?ref=v0.3.2"
-  pattern               = "/session"
-  positional_constraint = "STARTS_WITH"
-  metric_name           = "WAFRBRuleMatchSession"
+  allowed_headers = local.allowed_headers
+  metric_name     = "WAFRBRuleMatchSession"
```
Comment (x4121) [inline README.md:43]: Please fix the indentation manually until we have a pre-commit hook implemented for this repo

File: rate_based_rule/main.tf [hunk 2]
```diff
@@ -1,19 +1,24 @@
...
     }
   }
 }
 
 resource "aws_wafregional_rate_based_rule" "url_match_rule" {
-  name        = "wafregional_rate_based_match"
+  name  = "wafregional_rate_based_match"
+  count = length(var.allowed_headers) >= 1 ? 1 : 0
   tags        = local.tags
   metric_name = var.metric_name
```
Comment (x4121) [inline rate_based_rule/main.tf:23]: same here

**Review by x4121 — CHANGES_REQUESTED** [review]
Otherwise great PR, thanks :+1:

File: rate_based_rule/main.tf (hunk 2 above)
Comment (parmeet4dev) [reply rate_based_rule/main.tf]: updated the changes.
Thanks for noticing

**Review by x4121 — APPROVED** [review]

**Review by x4121 — CHANGES_REQUESTED** [review]
Ah sorry, just remembered: Please update the main Readme with the correct description and changelog

**Comment by parmeet4dev** [comment]
> Ah sorry, just remembered: Please update the main Readme with the correct description and changelog

updated the readme

**Comment by parmeet4dev** [comment]
Updated the main README.md file with the example and changelog

**Review by x4121 — APPROVED** [review]
The version tags in the Readme were not updated, fixed that.

**Review by x4121 — CHANGES_REQUESTED** [review]

**Comment by parmeet4dev** [comment]
1. updated main.tf to accommodate byte_match_set in rate-based rule based on the count.index
2. updated output.tf to only provide rule id
3. updated locals.tf to add tags

File: rate_based_rule/locals.tf [hunk 3]
```diff
@@ -1,3 +1,10 @@
+
 locals {
-  tags = var.tags
-}
+  tags = {
+    "env"         = "dev"
+    "squad"       = "inf"
+    "product"     = "internal"
+    "type"        = "infra"
+  }
```
Comment (x4121) [inline rate_based_rule/locals.tf:8]: Why did you add this? The tags should be passed into the module. Otherwise it will always be `squad = inf, env = dev` no matter who uses it

**Review by x4121 — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 4]
```diff
@@ -0,0 +1,4 @@
+.terraform/
+plan*
+*.tfstate
+*.backup
```
Comment (x4121) [inline .gitignore:4]: Did you add this to the repo by accident?

**Review by x4121 — APPROVED** [review]
